examples/server/server.cpp

```diff
@@ -2591,11 +2592,22 @@ int main(int argc, char ** argv) {
         return false;
     };
 
-    auto middleware_server_state = [&res_error, &state](const httplib::Request &, httplib::Response & res) {
+    auto middleware_server_state = [&res_error, &state](const httplib::Request & req, httplib::Response & res) {
         server_state current_state = state.load();
         if (current_state == SERVER_STATE_LOADING_MODEL) {
-            res_error(res, format_error_response("Loading model", ERROR_TYPE_UNAVAILABLE));
-            return false;
+            httplib::Request & modified_req = (httplib::Request &) req;
+            const char* path_c = modified_req.path.c_str();
+            int path_c_len = strlen(path_c);
+            char last_five[6];
+            strcpy(last_five, path_c + (path_c_len -5));
```

Comment:
what happen if `(path_c_len -5)` is negative?

Reply:
Better solution not using `c_str()` would be:

```
std::string path = req.path; // for example, "abc.html"
auto tmp = string_split(path, '.'); // "abc.html" --> {"abc", "html"}
std::string extension = tmp.back(); // == "html"
```

It also works if `req.path` is an empty string

Reply:
This isn't needed anymore since I'm simply calling res.set_content() as suggested [here](https://github.com/ggerganov/llama.cpp/pull/9401#discussion_r1751423407), which is a much cleaner solution.